pkg/mcs/tso/server/server.go

```diff
@@ -316,7 +323,74 @@ func (s *Server) initClient() error {
 	return err
 }
 
-func (s *Server) startServer() (err error) {
+func (s *Server) startGRPCServer(l net.Listener) {
+	defer s.serverLoopWg.Done()
+
+	gs := grpc.NewServer()
+	s.service.RegisterGRPCService(gs)
+	err := gs.Serve(l)
+	if err != cmux.ErrListenerClosed {
+		log.Fatal("grpc server stops serving unexpectedly.", errs.ZapError(err))
+	}
+	log.Info("gRPC server stop serving.", errs.ZapError(err))
+
+	// Attempt graceful stop (waits for pending RPCs), but force a stop if
+	// it doesn't happen in a reasonable amount of time.
+	done := make(chan struct{})
+	go func() {
+		gs.GracefulStop()
+		close(done)
+	}()
+	select {
+	case <-done:
+	case <-time.After(defaultGRPCGracefulStopTimeout):
+		log.Info(fmt.Sprintf("Stopping grpc gracefully is taking longer than %v."+
```

Comment:
how about `log.Info("stopping grpc gracefully is taking longer than expected and force stopping now.", zap.Duration("default", defaultGRPCGracefulStopTimeout))`

Reply:
changed.